medcat/utils/checkpoint.py

```diff
@@ -60,65 +54,118 @@ def count(self) -> int:
         return self._count
 
     @property
-    def metadata(self) -> Optional[Dict]:
-        return self._metadata
+    def dir_path(self) -> str:
+        return self._dir_path
 
     @classmethod
-    def restore(cls, dir_path: str) -> "Checkpoint":
+    def from_last_training(cls, parent_dir_path: str) -> "Checkpoint":
+        if not os.path.isdir(parent_dir_path):
+            raise ValueError(f"Checkpoint folder passed in does not exist: {parent_dir_path}")
+        ckpt_parent_folders = os.listdir(parent_dir_path)
+        if not ckpt_parent_folders:
+            raise ValueError("No existing training found")
+        ckpt_parent_folders.sort()
+        ckpt_file_path = os.path.abspath(os.path.join(parent_dir_path, ckpt_parent_folders[-1]))
+        checkpoint = cls.load(ckpt_file_path)
+        return checkpoint
+
+    @classmethod
+    def load(cls, dir_path: str) -> "Checkpoint":
+        r'''
+        Load the latest checkpoint from a directory.
+
+        Args:
+            dir_path (string):
+                The path to the directory containing checkpoint files
+        Returns:
+            A new checkpoint object
+        '''
         if not os.path.isdir(dir_path):
             raise Exception("Checkpoints not found. You need to train from scratch.")
         ckpt_file_paths = cls._get_ckpt_file_paths(dir_path)
         if not ckpt_file_paths:
             raise Exception("Checkpoints not found. You need to train from scratch.")
-        metadata = cls._load_metadata(dir_path)
         latest_ckpt = ckpt_file_paths[-1]
         steps, count = cls._get_steps_and_count(latest_ckpt)
-        checkpoint = cls(dir_path, steps=steps, metadata=metadata)
+        checkpoint = cls(dir_path, steps=steps)
         checkpoint._file_paths = ckpt_file_paths
         checkpoint._count = count
+        cls.log.info(f"Checkpoint loaded from {latest_ckpt}")
         return checkpoint
 
-    def purge(self) -> None:
-        shutil.rmtree(self._dir_path)
-        os.makedirs(self._dir_path)
+    def restore(self) -> None:
+        r'''
+        Restore the latest checkpoint.
+        '''
+        if not os.path.isdir(self._dir_path):
+            raise Exception("Checkpoints not found. You need to train from scratch.")
+        ckpt_file_paths = self._get_ckpt_file_paths(self._dir_path)
+        if not ckpt_file_paths:
+            raise Exception("Checkpoints not found. You need to train from scratch.")
+        latest_ckpt = ckpt_file_paths[-1]
+        _, count = self._get_steps_and_count(latest_ckpt)
+        self._file_paths = ckpt_file_paths
+        self._count = count
+
+    def purge(self) -> List[str]:
```

Comment:
things that remove from the filesystem programmatically are kind of scary - mostly as user error of the API can be catastrophic, maybe consider removing and users can manually do cleanup themselves? 
If we want to only store a max number of checkpoints for a given training run then thats a separate option

Reply:
A similar point was raised by @w-is-h in the past. Thought it would be convenient for devs to free their disk space (for potentially large CDB) programmatically given it is now safer than the first version thanks to pattern matching against file names. Similar methods can be found in here but deprecated. Not sure if it is due to the same reason concerning you guys. 
https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/remove_checkpoint

Reply:
didn't we agree to remove purge? its deprecated from the tf API, and torch / hf etc.  doesn't have it. The slimmer the API the better :)